src/prisma/basePrisma.service.ts

```diff
@@ -31,29 +41,55 @@ const PASSTHROUGH_METHODS = [
 export abstract class BasePrismaService<T extends PrismaModelNames>
   implements OnModuleInit
 {
-  @Inject(PrismaService)
-  private readonly _prisma!: PrismaService
-
   protected readonly logger = new Logger(this.constructor.name)
 
-  protected constructor(protected readonly modelName: T) {}
-
+  // inject prisma service via param injection, so child does not have direct access
+  @Inject(PrismaService)
```

Comment:
I think, possibly, if we're able to find a way to have `BasePrismaService` extend the generated Prisma models we want, then more than likely we should be able to just do away with our wrapper global `PrismaService` altogether.